src/helpers.ts

```diff
@@ -430,3 +430,18 @@ export function replaceWhitespace(text: string): string {
         .replace(/ /g, '&nbsp;')
         .replace(/\t/, '&nbsp;&nbsp;');
 }
+
+/**
+ * Work around nonstandard Firefox behavior when downloading a PDF by changing
+ * a PDF mimetype to application/octet-stream.
+ *
+ * See https://github.com/threema-ch/threema-web/issues/1118
+ */
+export function firefoxWorkaroundPdfDownload(mimetype: string): string {
+    const uagent = window.navigator.userAgent.toLowerCase();
+    const isFirefox = /mozilla/.test(uagent) && /firefox/.test(uagent); // Ugh
+    if (isFirefox && mimetype === 'application/pdf') {
+        return 'application/octet-stream';
+    }
+    return mimetype;
```

Comment:
Why don't you use the existing BrowserInfo class?

Reply:
I assume you mean `BrowserService`? Because it's an angular service with dependencies on other services, and I didn't want to inject it everywhere (and pass it through to the helper). Additionally, we only need to detect Firefox, the `BrowserService` does more than that.

But the Firefox detection line is taken from the `BrowserService`.

Reply:
Ok, I don't have a strong opinion on this, just wondering.

Reply:
`BrowserInfo` is just the data holder returned by `BrowserService` 🙂